### PR TITLE
Eng 415 manual component bundle deletion

### PR DESCRIPTION
**Pull request by firegloves**
I have changed Content Template usage request to match the new one in entando-components jacms
Then I've unified the usage requests and fixed the tests
I've added a lot of tests because of Sonar that was blocking me for a too low coverage

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **firegloves**

If they are not already assigned, you can assign the PR to them by writing `/assign @firegloves` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando-k8s%2Fentando-component-manager).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando-k8s/entando-component-manager/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/main/java/org/entando/kubernetes/client/core/DefaultEntandoCoreClient.java [hunk 1]
```diff
@@ -1,26 +1,8 @@
...
-import org.entando.kubernetes.model.entandocore.EntandoCoreComponentUsage;
-import org.entando.kubernetes.model.entandocore.EntandoCoreContentModel;
-import org.entando.kubernetes.model.entandocore.EntandoCoreFile;
-import org.entando.kubernetes.model.entandocore.EntandoCoreFolder;
-import org.entando.kubernetes.model.entandocore.EntandoCoreFragment;
-import org.entando.kubernetes.model.entandocore.EntandoCorePage;
-import org.entando.kubernetes.model.entandocore.EntandoCorePageTemplate;
-import org.entando.kubernetes.model.entandocore.EntandoCoreWidget;
+import org.entando.kubernetes.model.bundle.descriptor.*;
+import org.entando.kubernetes.model.entandocore.*;
```
Comment (ffleandro) [inline src/main/java/org/entando/kubernetes/client/core/DefaultEntandoCoreClient.java:5]: These imports are out of Entando's code styles and will generate lot's of unnecessary changes in the next PRs.
Please add the intellij codestyle and organize imports on the whole project before creating a PR.

**Entando Code Style:**
https://github.com/entando/entando-code-style

File: src/test/java/org/entando/kubernetes/client/EntandoCoreClientTest.java [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
+import org.springframework.web.client.HttpClientErrorException;
+
+import java.util.Arrays;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 
-@Tag("in-process")
-public class EntandoCoreClientTest {
+@Tag("unit")
```
Comment (ffleandro) [inline src/test/java/org/entando/kubernetes/client/EntandoCoreClientTest.java:20]: Are you sure about this?

Comment (firegloves) [reply src/test/java/org/entando/kubernetes/client/EntandoCoreClientTest.java:31]: Yes, that class's tests were not executed. If you look at the pom you can see that in this project we execute only _unit_ and _component_ tagged tests

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando-k8s/entando-component-manager/pull/49#pullrequestreview-455807480" title="Approved">ffleandro</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando-k8s%2Fentando-component-manager).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/entando-k8s/entando-component-manager/blob/master/OWNERS)~~ [ffleandro]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>
